app/lib/presentation/pages/splash_screen_page.dart

```diff
@@ -25,22 +25,32 @@ class _SplashPageState extends State<SplashPage> {
     RandomIdController.instance.init();
     final int loginCounter =
         PreferencesController.instance.getInt(PreferenceKeys.loginCounter) ?? 0;
+    final bool isFirstLogin = loginCounter == 0;
     PreferencesController.instance
         .setInt(PreferenceKeys.loginCounter, loginCounter + 1);
     CollectionsObject.init();
     AnswersCollectionsObject.init();
     TtsController.instance.init();
-    _navigate();
+    _navigate(isFirstLogin);
   }
 
-  Future _navigate() async {
+  Future _navigate(bool isFirstLogin) async {
     Navigator.of(context).pop();
 
-    Navigator.of(context).push(
-      MaterialPageRoute(
-        builder: (context) => WelcomePage(),
-      ),
-    );
+    if (isFirstLogin) {
```

Comment:
Checking only first login will create a bug where if user close the app before finishing the itrudication screen it will skip to the home page.
Add another variable
```dart
    final bool finishedIntroduction = PreferencesController.instance
            .getBool(PreferenceKeys.finishedIntroduction) ??
        false;
```